core/node/node_framework/src/implementations/layers/eth_sender/aggregator.rs

```diff
@@ -93,7 +97,13 @@ impl WiringLayer for EthTxAggregatorLayer {
         let master_pool = input.master_pool.get().await.unwrap();
         let replica_pool = input.replica_pool.get().await.unwrap();
 
-        let eth_client = input.eth_client.0;
+        let (eth_client, settlement_mode) = match (input.eth_client, input.eth_client_l2) {
+            (Some(l1_client), None) => (l1_client.0, SettlementMode::SettlesToL1),
+            (None, Some(l2_client)) => (l2_client.0, SettlementMode::Gateway),
+            (_, _) => Err(WiringError::Configuration(
+                "Aggregator requires either l1 or l2 client to operatate".to_string(),
```

Comment:
what happens if both of those are provided?

Reply:
I've changed for not so that it relies on gas_adjuster settlement_mode for which client to chose.